### PR TITLE
Implemented matchmakers in the market

### PR DESCRIPTION
**Pull request by devos50**
This PR implements various functionalities in the market community:
- **generic matching**: instead of matching orders, it is now possible to match any tick with any other tick (first commit).
- **matchmakers**: matchmakers are responsible for finding matches between asks and bids in the market. By default, every new user in a matchmaker, however, being matchmaker can be disabled with a configuration option (second commit).
- **test suite improvement**: I've added some additional tests and fixed existing ones (third commit). 

Matchmaking protocol:

matching

### CODE REVIEW

**Comment by devos50** [comment]
retest this please

File: Tribler/community/market/core/matching_engine.py [hunk 1]
```diff
@@ -28,236 +27,226 @@ def __init__(self, order_book):
 
         self.order_book = order_book
 
+    @staticmethod
+    def generate_match_id():
```
Comment (qstokkink) [inline Tribler/community/market/core/matching_engine.py]: Would it be a good idea to check if this `match_id` exists in the database before using it? Or do you take care of this in some other way?

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@qstokkink the Mac GUI test failed again (probably due to the window thingy) but other than that, I've addressed your issue and now, a unique match ID is always generated 👍 . This PR is ready for merge I guess?

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
Windows builder is very unstable, unit tests take 19 minutes to execute...

**Comment by devos50** [comment]
But I fixed the Pylint issue 👍

**Pull request merged**